drivers/docker/docklog/docker_logger.go

```diff
@@ -168,3 +186,37 @@ func (d *dockerLogger) getDockerClient(opts *StartOpts) (*docker.Client, error)
 
 	return newClient, merr.ErrorOrNil()
 }
+
+func isLoggingTerminalError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	terminals := []string{
+		"configured logging driver does not support reading",
+	}
+
+	for _, c := range terminals {
+		if strings.Contains(err.Error(), c) {
```

Comment:
any particular reason you decide to look at the error message instead of the status code (501)?

Reply:
I can check status code too, but I wanted to be conservative in what would be considered a terminal error.

Reply:
`501 Not Implemented` seems likely to be a permanent error on the container logs endpoint of the docker API. My only concern would be if the content of the error message changed in a future version of the Docker API (although, the same could happen with the 501 error; it's not even documented as a potential return code.)

I'll leave this to your judgement.

Reply:
Yeah. I added handling 501 as well in a follow up commit. Another concern about status checking is that if the error gets wrapped somewhere, we’d lose the type and need to check error string.

Now we check both the status code as well as the error message.